csm/core/services/usl.py

```diff
@@ -484,21 +485,39 @@ async def get_system_certificates_by_type(self, material_type: str) -> bytes:
             raise CsmNotFoundError(reason)
         return material
 
-    async def get_network_interfaces(self) -> List[Dict[str, Any]]:
+    async def _get_public_ip(self) -> str:
         """
-        Provides a list of all network interfaces in a system.
+        Reads UDS public IP from global index in an attempt to override UDS default behavior.
+        If it is not found, uses cluster IP as UDS public IP.
 
-        :return: A list containing dictionaries, each containing information about a specific
-            network interface.
+        :return: A string representing UDS public IP address.
         """
+        try:
+            ip = Conf.get(const.CSM_GLOBAL_INDEX, 'UDS.public_ip')
+            ip_address(ip)
+            return ip
+        except ValueError as e:
+            reason = 'UDS public IP override failed---following usual code path'
+            Log.debug(f'{reason}. Error: {e}')
```

Comment:
Log this as error.

Reply:
I don't think we should. The IP address override is an optional feature. In most scenarios it is expected that we will fall into the next use case, that is, reading cluster IP info from the provisioner.

I though that logging that highest levels (even `info`) might include some undesired verbosity in the logs. People will think that this is something that shouldn't fail when, in fact, it will in most cases.